src/objects/BatchedMesh.js

```diff
@@ -832,6 +832,35 @@ class BatchedMesh extends Mesh {
 
 	}
 
+	setGeometryIdAt( instanceId, geometryId ) {
+
+		// return early if the geometry is out of range or not active
+		const drawInfo = this._drawInfo;
+		if ( instanceId >= drawInfo.length || drawInfo[ instanceId ].active === false ) {
+
+			return null;
+
+		}
+
+		drawInfo[ instanceId ].geometryIndex = geometryId;
```

Comment:
We should block the user from setting an invalid geometryId, I think.

Reply:
I added test and a warning for the user, also currently setMatrixAt and other function can return null without warning and maybe it can be good to add warning as well for them in another PR ?

I'm not sure of the current warning policy in Threejs but it can avoid a lot of headaches for devs and issues opening  to avoid silent error.

Reply:
I'll leave whether we should add warnings to someone else. An error with a message indicating what went wrong may be best but that can left to another PR if it gets added.

Reply:
I just removed the warning to keep consistency with the rest of the class and i'll open a discussion about it separately. thanks!